tests/catch/unit/event/Unit_hipEventQuery.cc

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <hip_test_common.hh>
+
+// Wait till *ptr is equal to old_val, replace it with new_val
+// Basically wrapper around CAS
+__global__ void blockKernel(int* ptr, int old_val, int new_val) {
+  while (atomicCAS_system(ptr, old_val, new_val) != new_val)
+    ;
+}
+
+TEST_CASE("Unit_hipEventQuery_DifferentDevice") {
+  if (HipTest::getDeviceCount() <= 1) {
+    INFO("needs atleast 2 devices to run");
+    return;
+  }
+
+  hipEvent_t event1, event2;
+  HIP_CHECK(hipEventCreate(&event1));
+  HIP_CHECK(hipEventCreate(&event2));
+  REQUIRE(event1 != nullptr);
+  REQUIRE(event2 != nullptr);
+
+  hipStream_t stream{nullptr};
+  HIP_CHECK(hipStreamCreate(&stream));
+  REQUIRE(stream != nullptr);
+
+  int* ptr{nullptr};
+  HIP_CHECK(hipMallocManaged(&ptr, sizeof(int), hipMemAttachGlobal));
+  REQUIRE(ptr != nullptr);
+
+  // Set it to 0
+  __atomic_store_n(ptr, 0, __ATOMIC_SEQ_CST);
+
+  constexpr int old_val = 10, new_val = 20;
+
+  // Start kernel on 1st device
+  {
```

Comment:
This part of the test should run even when there is only one device

Reply:
done